src/caret_analyze/architecture/struct/node.py

```diff
@@ -186,6 +187,48 @@ def to_value(self) -> NodeStructValue:
             None if self.variable_passings is None
             else tuple(v.to_value() for v in self.variable_passings))
 
+    def assign_message_context(self, context_type, sub_topic_name: str, pub_topic_name: str):
+        try:
+            path = self.get_path(sub_topic_name, pub_topic_name)
+            path.message_context =\
+                MessageContextStruct.create_instance(context_type, {},
+                                                     self.node_name, path.subscription,
+                                                     path.publisher, None)
+        except ItemNotFoundError:
+            subscription: SubscriptionStruct = \
+                Util.find_one(lambda x: x.topic_name == sub_topic_name, self._subscriptions)
+            publisher: PublisherStruct = \
+                Util.find_one(lambda x: x.topic_name == pub_topic_name, self._publishers)
+            message_context =\
+                MessageContextStruct.create_instance(context_type, {},
+                                                     self.node_name, subscription,
+                                                     publisher, None)
+            path = NodePathStruct(self.node_name, subscription, publisher,
+                                  None, message_context)
+            self._node_paths.append(path)
+
+    def assign_publisher(self, pub_topic_name: str, callback_function_name: str):
+        callback: CallbackStruct = \
+            Util.find_one(lambda x: x.callback_name == callback_function_name, self.callbacks)
```

Comment:
Change of commit [93f96b9](https://github.com/tier4/CARET_analyze/pull/196/commits/93f96b9a54029c50c1d9392d5c2748cfb8ab3e3c).
Search callback in NodeStruct::assign_publisher not in Architecture::assign_publisher.

Reply:
LGTM